Package.swift

```diff
@@ -28,30 +28,51 @@ let package = Package(
         .watchOS(.v6)
     ],
     products: [
+        .library(name: "Smithy", targets: ["Smithy"]),
```

Comment:
I am not currently sure about this module's name and exact scope... this is intended to be things that Smithy uses across all modules.

I might break it down further though to have such modules for clients and for models, since it seems some smithy-swift customers use them separately.

I'd like to leave it for now, and we can revisit it later in modularization.